src/Vector.cpp

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include "Vector.h"
+
+
+Vector::Vector(const size_t& size, const int& value)
+    : _size(0), _capacity(0), _data(nullptr) {
+    for (size_t i = 0; i < size; i++) {
+        push_back(value);
+    }
+}
+
+Vector::~Vector() {
+    if (_data != nullptr)
+        delete[] _data;
+}
+
+// Copy constructor
+Vector::Vector(const Vector& other)
+    : _size(0), _capacity(0), _data(nullptr) {
+    *this = other;
+}
+
+Vector& Vector::operator=(const Vector& other) {
+    if (this == &other)
+        return *this;
+
+    for (int i = 0; i < other.size(); i++) {
```

Comment:
Won't this append `other`'s elements to `this`'s array? Instead of replacing it like a `copy-assignment` operator should